service_crategen/src/codegen/generator/mod.rs

```diff
@@ -212,6 +212,9 @@ fn mutate_type_name(type_name: &str) -> String {
         // EC2 has a CancelSpotFleetRequestsError struct, avoid collision with our error enum
         "CancelSpotFleetRequests" => "EC2CancelSpotFleetRequests".to_owned(),
 
+        // Support has a conveniently names "Result" type
+        "Result" => "SupportResult".to_owned(),
```

Comment:
Just so this doesn't get lost on the sea of new service PRs...

This service requires a codegen change to rename a type so it doesn't collide with the global Rust `Result` type.